### PR TITLE
Improve error code lookup

### PR DESCRIPTION
**Pull request by nbougalis**
The existing implementation of `transResultInfo` uses a naive linear search. Use a `std::unordered_map` instead.

Remove already triggered ledger switches.

### CODE REVIEW

File: src/ripple/protocol/impl/TER.cpp [hunk 1]
```diff
@@ -19,141 +19,133 @@
 
 #include <BeastConfig.h>
 #include <ripple/protocol/TER.h>
+#include <map>
+#include <type_traits>
```
Comment (seelabs) [inline src/ripple/protocol/impl/TER.cpp]: Needs `#include<unordered_map>` for non-unity build.

File: src/ripple/app/tx/impl/Transactor.cpp [hunk 2]
```diff
@@ -694,14 +679,15 @@ Transactor::operator()()
...
+        // at view() past this point.
     }
 
-    auto const result =
-        std::make_pair(terResult, didApply);
-    log(result, j_);
-    return result;
+    JLOG(j_.trace) <<
+        "apply: " << transToken(terResult) <<
+        ", " << (didApply ? "true" : "false");
```
Comment (seelabs) [inline src/ripple/app/tx/impl/Transactor.cpp]: Could use `std::boolalpha`, though that does change that format flag for future calls. Could change it back with `std::noboolalpha`.

File: src/ripple/protocol/impl/TER.cpp [hunk 3]
```diff
@@ -19,141 +19,133 @@
 
 #include <BeastConfig.h>
 #include <ripple/protocol/TER.h>
+#include <map>
```
Comment (seelabs) [inline src/ripple/protocol/impl/TER.cpp]: `#include <map>` is not needed in this file.

**Comment by seelabs** [comment]
:+1: After the non-unity build problem is taken care of.

**Comment by ximinez** [comment]
Reviewers: @ximinez @seelabs @vinniefalco

**Comment by ximinez** [comment]
Looks good. :+1:

**Comment by nbougalis** [comment]
Merged as 298ef4ac4d379628a957a6cddeb6b38e6992b80c and 0633ef1ba1ee47d62c8026a47c96f99e513ad618.

**Pull request closed without merging**